.github/workflows/node.js.yml

```diff
@@ -14,46 +14,108 @@ env:
+
+  #     - name: Deploy
+  #       if: github.ref == 'refs/heads/master'    
+  #       run: |
+  #         git remote set-url origin git@github.com:yaptide/web_dev.git
+  #         npm run deploy -- -u "github-actions-bot <support+actions@github.com>" -r "git@github.com:yaptide/web_dev.git"    
+
+  deploy:
     runs-on: ubuntu-latest
-    needs: [build]    
+    # needs: [test_and_deploy]
+    env:
+      FRONTEND_HOST: 149.156.182.181
+      FRONTEND_USER: ubuntu
+
     steps:
-      - name: Checkout 🛎️
+      # get the repo to the github agent
+      - name: Checkout
         uses: actions/checkout@v2
+        with:
+          submodules: true
 
-      - name: Install 🔧 
-        run: npm install
-
-      - name: webfactory/ssh-agent
-        if: github.ref == 'refs/heads/master'    
-        uses: webfactory/ssh-agent@v0.5.3
+      # add keys to the SSH agent
+      - uses: webfactory/ssh-agent@v0.5.3
+        # if: github.ref == 'refs/heads/master'
         with:
-          # Private SSH key to register in the SSH agent
-          ssh-private-key: ${{ secrets.WEBDEV }}
+          ssh-private-key: |
+            ${{ secrets.DEPLOY_C3_TEST }}
 
-      - name: Deploy
-        if: github.ref == 'refs/heads/master'    
+      # Create the known_hosts file with the open stack machines public keys
+      - name: Add host to known_hosts
         run: |
-          git remote set-url origin git@github.com:yaptide/web_dev.git
-          npm run deploy -- -u "github-actions-bot <support+actions@github.com>" -r "git@github.com:yaptide/web_dev.git"    
+          mkdir -p ~/.ssh
+          chmod 700 ~/.ssh
+          ssh-keyscan -H ${{ env.FRONTEND_HOST }} >> ~/.ssh/known_hosts
+
+      # kill docker containers, then remove them
+      - name: Kill our docker containers
+        continue-on-error: true
+        # if: github.ref == 'refs/heads/master'
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ env.FRONTEND_HOST }}
+          username: ${{ env.FRONTEND_USER }}
+          key: ${{ secrets.DEPLOY_C3_TEST }}
+          script: |
+            docker stop yaptide-nginx
+            docker system prune
+            docker rmi yaptide-nginx
     
+      # copy the new files form the github agent onto the machine
+      - name: Copy files to the machine
+      
+        # if: github.ref == 'refs/heads/master'
+        run: rsync -e ssh -A ${{ env.FRONTEND_USER }}@${{ env.FRONTEND_HOST }}:~/ui --delete --recursive
+
+      # No need to run these processes using screen, as docker-compose with --detach option sends the process to the background
+      - name: Compose docker images and run them
+        # if: github.ref == 'refs/heads/master'
+        uses: appleboy/ssh-action@master
+        with:
+          host: ${{ env.FRONTEND_HOST }}
+          username: ${{ env.FRONTEND_USER }}
+          key: ${{ secrets.DEPLOY_C3_TEST }}
+          command_timeout: 30m
+          script: |
+            cd ui
+            npm install
```

Comment:
Which version of node we need at the frontend machine ?

![image](https://user-images.githubusercontent.com/7374733/142022266-4bd462b9-de40-4135-8db1-f14aa75d5a98.png)

Reply:
Consider replacing this with Docker one-liner

Reply:
Node and nmp updated 
![obraz](https://user-images.githubusercontent.com/26521377/142450540-42c4366e-dd11-4804-8b1b-a88f91b4e426.png)

Moved to issue https://github.com/yaptide/ui/issues/264